src/test/java/com/codejoust/main/api/ProblemTests.java

```diff
@@ -491,4 +425,64 @@ public void deleteProblemTagSuccess() throws Exception {
 
         assertEquals(problemTag, problemTagReturn);
     }
+
+    @Test
+    public void deleteProblemWithExistingTagsSuccess() throws Exception {
+        /**
+         * 1. Create two problems with the same tag (tag should be reused)
+         * 2. Delete one of the problems successfully
+         * 3. The tag should remain in the database
+         */
```

Comment:
Thanks for adding these explanations! I think these are good standard practice going forward 👍